platform/server-config/src/main/java/org/motechproject/server/config/SettingsFacade.java

```diff
@@ -215,6 +221,53 @@ public void saveRawConfig(String filename, String jsonText) {
     }
 
     /**
+     *  Allows persisting of raw JSON properties either in the database or file.
+     *
+     * @param filename json filename
+     * @param properties properties to persist
+     * @throws org.motechproject.commons.api.MotechException when I/O error occurs
+     */
+    public void saveRawConfig(String filename, Properties properties) {
+        Gson gson = new GsonBuilder().create();
+        Map<String, String> map = new HashMap<String, String>();
+        for (Entry<Object, Object> prop : properties.entrySet()) {
+            map.put((String) prop.getKey(), (String) prop.getValue());
+        }
+        saveRawConfig(filename, gson.toJson(map));
+    }
+
+    /**
+     * Allows to retrieve raw JSON data either from the database or file.
+     *
+     * @param filename Resource filename
+     * @return Raw JSON data as InputStream
+     * @throws org.motechproject.commons.api.MotechException when I/O error occurs
+     */
+    public Properties getPropertiesFromRawConfigFile(String filename) {
```

Comment:
Again, this seems pointless parsing to the desired format should be left to the caller